libs/content-watcher-lib/src/scanner/scanner.service.ts

```diff
@@ -111,21 +114,27 @@ export class ScannerService implements OnApplicationBootstrap, OnApplicationShut
 
         const queueSize = await this.ipfsQueue.count();
         if (queueSize > this.config.queueHighWater) {
-          this.logger.log('Queue soft limit reached; pausing scan until next interval');
+          const now = Date.now();
+          if (now - this.lastQueueLimitLogTime > this.LOG_THROTTLE_MS) {
+            this.logger.log(
+              `Queue soft limit reached (${queueSize}/${this.config.queueHighWater}); pausing scan until next interval`,
+            );
+            this.lastQueueLimitLogTime = now;
+          }
```

Comment:
Another way to do this would be to introduce a class variable `isHighWater`. Set true when soft limit reached, false if queue has drained enough.

This way we can log when we reach high water, but we don't have to keep logging while we're still in that state. But if we drain & reach high water again, we'll still get the log.

It can be useful in the log to know when we transition between the states, but not keep logging while we're in the state.